pkg/apis/wsecurity/v1/count.go

```diff
@@ -0,0 +1,146 @@
+package v1
+
+import (
+	"bytes"
+	"fmt"
+)
+
+//////////////////// CountProfile ////////////////
+
+// Exposes ValueProfile interface
+type CountProfile uint8
+
+func (profile *CountProfile) Profile(args ...interface{}) {
+	*profile = CountProfile(args[0].(uint8))
+}
+
+func (profile *CountProfile) String(depth int) string {
+	return fmt.Sprintf("%d", uint8(*profile))
+}
+
+//////////////////// CountPile ////////////////
+
+// Exposes ValuePile interface
+type CountPile []uint8
+
+func (pile *CountPile) Add(valProfile ValueProfile) {
+	profile := *valProfile.(*CountProfile)
```

Comment:
The function signatures are all defined in v1.go
Same signatures will be used for other data types

Reply:
Again, I think using generics will simplify this, and also make it harder to accidentally `CountPile.Add(StringProfile)`, which looks like it will explode with a `panic` at the moment.

Reply:
As discussed above, each method is implemented differently on each data type,  in this case, do you still see a way for generics to help remove the runtime type assertions?

It seems that even when we use generics, we will still need to have different type assertions for each implementation, simply because each of them does something different with the data... so:

```golang
func (p *KeyValPile) Add(...) // need to assert that the data is of type KeyValProfile}
func (p *ReqPile) Add(...) // need to assert that the data is of type ReqProfile
```



See above - I can't see how generics help us in simplifying or removing any code.

Reply:
If `KeyValPile` implements `Pile[KeyVal]`, and `ReqPile` implements `Pile[Req]`, then they would have different arguments, but implement the same interface.

Reply:
@evankanderson  - what generics let you do is have the same Add function (same lines of code) implemented once for all dataTypes. Since we need different Add functions (different lines of code for each function, regardless of the type being added) implementing one generic `func Pile[T] Add()` function will not help us.

Reply:
Generics added to TODO under Future